staging/src/k8s.io/apimachinery/pkg/runtime/converter.go

```diff
@@ -816,10 +819,14 @@ func structToUnstructured(sv, dv reflect.Value) error {
 			// This field should be skipped.
 			continue
 		}
-		if fieldInfo.omitempty && isZero(fv) {
+		if fieldInfo.omitempty && isEmpty(fv) {
 			// omitempty fields should be ignored.
 			continue
 		}
+		if fieldInfo.omitzero && fv.IsZero() {
```

Comment:
Is there a test case with an omitzero field whose type implements `interface{ IsZero() bool }`? I would have expected a parallel implementation of the logic from `encoding/json`.

Reply:
good call, I hoisted in the custom IsZero() tests from go and reworked this to make sure they all work identically here